### ISSUE TITLE
Dropdowns as indirect descendant of dropdowns

### ISSUE THREAD

**Issue by no-oyvind** [radix-ui/primitives#1016]
## Bug report

_Sorry for the confusing title. I've attached a sandbox to help clear things up._

I'm trying to create a reusable dropdown menu item which manages its own dialog. It works fairly well, but – if the dialog contains a dropdown menu itself, the dropdown menu isn't rendered.

### Current Behavior

Any Dropdown menus contained by a Dialog.Root, which is a descendant of a Dropdown.Root, are not rendered

```
Dropdown
    Dialog
        Dropdown (not rendered)
```

### Expected behavior

Any Dropdown menus contained by the Dialog are rendered and usable like any other Dropdown menu.

```
Dropdown
    Dialog
        Dropdown (rendered)
```

### Reproducible example

The issue is illustrated in this sandbox:
https://codesandbox.io/s/test-radix-dialog-forked-4j9n8?file=/src/App.tsx

### Suggested solution

_disclaimer: I might have completely misunderstood what the problem is, so a grain of salt is recommented_

I can imagine the use-case is complicated by the way we already define nested menus with nested Roots and TriggerItems. 

I'm not familiar with the implementation, but possibly by providing a way to explicitly tell the current Dropdown context to ignore a subtree? In pigeon Radix: `<Dropdown.Ignore><DialogWithDropdown /></Dropdown.Ignore>`

```tsx
const ItemWithModal = React.forwardRef<HTMLDivElement, Dropdown.MenuItemProps>(
  (props, ref) => {
    const [isOpen, setIsOpen] = React.useState(false);

    const openModalAndKeepDropwdownOpen = (e: Event) => {
      e.preventDefault();
      setIsOpen(true);
    };

    return (
      <>
        <Dropdown.Item
          {...props}
          ref={ref}
          onSelect={openModalAndKeepDropwdownOpen}
        >
          Open modal
        </Dropdown.Item>
        {/* Tell the parent dropdown to ignore/skip this subtree */}
        <Dropdown.Ignore>
          <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
            <DialogOverlay />
            <DialogContent>
              This should be followed by a dropdown:
              <Dropdown.Root>
                <Dropdown.Trigger>Nested dropdown trigger</Dropdown.Trigger>
                <Dropdown.Content>
                  <Dropdown.Item>Dropdown content</Dropdown.Item>
                </Dropdown.Content>
              </Dropdown.Root>
            </DialogContent>
          </Dialog.Root>
        </Dropdown.Ignore>
        {/* End ignored/skipped subtree */}
      </>
    );
  }
);
```

### Additional context

It's important to note that the `ItemWithModal` component in the sandbox must manage its own dialog. Pulling the dialog out as a sibling of the outer Dropdown works, but defeats the purpose of having a reusable menu item which opens a dialog.

### Your environment

It's all described in the sandbox, but

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) | @radix-ui/react-dialogs        | 0.1.1        |
| Radix Package(s) | @radix-ui/react-dropwdown-menus        | 0.1.1        |
| React            | n/a     | 17.0.2        |
| Browser          | Chrome        |  96.0.4664.55        |
| Browser          | Safari        | 14.1.1       |
| Assistive tech   |         |         |
| Node             | n/a     |         |
| npm/yarn         |         |         |
| Operating System | MacOS        | 11.4        |

**Comment by jjenzz** [comment]
Thanks for this @no-oyvind. 

I imagine you're right that the dropdown believes the second menu is a sub menu.

I wonder if our API would have been better off as `DropdownMenu.Sub` or something after all 😬

**Comment by no-oyvind** [comment]
Yeah, that would probably be easier to manage, in hindsight. But I presume it would also break the API by now. 

I'm not going to promote the `<Ignore />` approach. It feels like treating symptoms rather than fixing the root cause. I just wanted to add that it could be used for optimisation if we have deeply nested doms we don't want Radix to traverse. I have no idea how Radix interacts with the DOM, though. And it can encourage unnecessary nesting, which we might want to avoid.

I'd love to help out on fixing this, but there are too few hours in a day the next couple of months. This issue is blocking us from adapting Radix for much more than basic tooltips and radio buttons, though. Any thoughts on when or if this can be looked into? I can imagine you're already up to your ears in fixes and features.

**Comment by jjenzz** [comment]
> But I presume it would also break the API by now.

This is the perfect time to introduce breaking API changes as we are not version 1 yet. @andy-hook Do you think that exposing a separate `DropdownMenu.Sub` part would solve this? We could maybe even remove `DropdownMenu.TriggerItem` then I think and re-use `DropdownMenu.Trigger`.

**Comment by andy-hook** [comment]
> @andy-hook Do you think that exposing a separate `DropdownMenu.Sub` part would solve this? We could maybe even remove `DropdownMenu.TriggerItem` then and re-use `DropdownMenu.Trigger` for sub menu triggers 🤔

Think so, would need to look at the trigger code again to see how feasible the second suggestion is. I remember us having some issues there which facilitated need for the extra part.

**Comment by benoitgrelard** [comment]
Closed as tracked by #1309
